build/arch.mk

```diff
@@ -30,14 +30,24 @@ CFLAGS += -DHAVE_NEON_AARCH64
 endif
 endif
 
-#for loongson
+#for mips
 ifneq ($(filter mips mips64, $(ARCH)),)
 ifeq ($(USE_ASM), Yes)
 ASM_ARCH = mips
 ASMFLAGS += -I$(SRC_PATH)codec/common/mips/
-LOONGSON3A = $(shell g++ -dM -E - < /dev/null | grep '_MIPS_TUNE ' | cut -f 3 -d " ")
-ifeq ($(LOONGSON3A), "loongson3a")
-CFLAGS += -DHAVE_MMI
+#mmi
+ifeq ($(ENABLE_MMI), Yes)
+ENABLE_MMI = $(shell $(SRC_PATH)build/mips-simd-check.sh $(CC) mmi)
+ifeq ($(ENABLE_MMI), Yes)
+CFLAGS += -march=loongson3a -DHAVE_MMI
```

Comment:
-Wa,-mloongson-mmi,-mloongson-ext will be better as it won't generate Loongson instructions in C part of generic MIPS Build.

Reply:
Lower versions of GCC(like 4.9.3 on loongson) are not support -mloongson-mmi,-mloongson-ext. Maybe we can only add -march=loongson3a on *_mmi.c.

Reply:
As far as I know, Loongson had backport this option to GCC 4.9 in Loongnix. And it also mainlined in GCC9.

So can we just drop support for old compilers in upstream code?